src/cached_block_device.rs

```diff
@@ -186,6 +186,10 @@ impl<B: BlockDevice> BlockDevice for CachedBlockDevice<B> {
         Ok(())
     }
 
+    fn flush(&mut self) -> Result<(), Self::Error> {
+        self.block_device.flush()
```

Comment:
You must flush the LRU !

At least flush all dirty blocks in the LRU. You should also consider emptying out the LRU all together.